docs/user-guide.md

```diff
@@ -49,9 +49,10 @@ Usage of kube-router:
       --disable-source-dest-check        Disable the source-dest-check attribute for AWS EC2 instances. When this option is false, it must be set some other way. (default true)
       --enable-cni                       Enable CNI plugin. Disable if you want to use kube-router features alongside another CNI plugin. (default true)
       --enable-ibgp                      Enables peering with nodes with the same ASN, if disabled will only peer with external BGP peers (default true)
-      --enable-overlay                   When enable-overlay set to true, IP-in-IP tunneling is used for pod-to-pod networking across nodes in different subnets. When set to false no tunneling is used and routing infrastrcture is expected to route traffic for pod-to-pod networking across nodes in different subnets (default true)
+      --enable-overlay                   When enable-overlay is set to true, IP-in-IP tunneling is used for pod-to-pod networking across nodes in different subnets. When set to false no tunneling is used and routing infrastructure is expected to route traffic for pod-to-pod networking across nodes in different subnets (default true)
       --enable-pod-egress                SNAT traffic from Pods to destinations outside the cluster. (default true)
       --enable-pprof                     Enables pprof for debugging performance and memory leak issues.
+      --full-overlay                     When full-overlay is set to true, it changes "--enable-overlay=true" default behavior so that IP-in-IP tunneling is used for pod-to-pod networking across nodes regardless of the subnet the nodes are in. When set to false, the default, default "--enable-overlay=true" behavior is used
```

Comment:
What do you think about using `--overlay-type` instead, with a default value `regional` or `subnet`. And the other option is `type`?

Reply:
Sounds like a good idea to me. I don't know if I understood your question correctly ("And the other option is type?"). Could you elaborate a bit more?

Reply:
Sorry, typo. I meant "And the other option is full". 

So user needs to set both
```
--enable-overlay=true
--overlay-type={subnet,full}
```

Reply:
cc @murali-reddy

Reply:
> Sorry, typo. I meant "And the other option is full".

Now I got it :)

> So user needs to set both
> 
> ```
> --enable-overlay=true
> --overlay-type={subnet,full}
> ```

Yes, it makes sense. I've updated the code to reflect that.